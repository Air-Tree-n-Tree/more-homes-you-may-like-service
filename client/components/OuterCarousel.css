.outerCarousel {
  border: 3px solid black;
  display: flex;  
  overflow: hidden;
  width: 900px;
}

.outerCarouselComponent {
  overflow: hidden;
  display: flex;
  align-items: center;
}

.outerCarouselItem {
  display: inline-block;
  width: 300px;
  height: 400px;
}

.slider {
  display: flex;
  align-content: flex-start;
}

.outerCarouselItem {
  flex-shrink: 0;
  width: 300px;
}

.outerLeftButton {
  font-size: 100px;
}

.outerLeftButton::after {
  content: "⇦";
}

.outerRightButton {
  font-size: 100px;
}

.outerRightButton::after {
  content: "⇨";
}
