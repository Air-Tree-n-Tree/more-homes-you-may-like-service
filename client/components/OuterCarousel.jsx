import React from 'react';
import _ from 'underscore';
import OuterCarouselItem from './OuterCarouselItem';
import './carousel.css';
import { Animate } from 'react-rebound';


class OuterCarousel extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentIndex: 0
    };
    this.clickHandlerLeft = this.clickHandlerLeft.bind(this);
    this.clickHandlerRight = this.clickHandlerRight.bind(this);
  }

  getOffset() {
    let { currentIndex } = this.state;
    return currentIndex * -345;
  }

  slideRight() {
    const { currentIndex } = this.state;
    const { listings } = this.props;
    if (currentIndex > listings.length-4) {
      this.setState({currentIndex: 0});
    } else {
      this.setState({currentIndex: currentIndex+1});
    }
  }

  slideLeft() {
    const { currentIndex } = this.state;
    const { listings } = this.props;
    if (currentIndex === 0) {
      this.setState({currentIndex: listings.length-3});
    } else {
      this.setState({currentIndex: currentIndex-1});
    }
  }

  clickHandlerRight(e) {
    this.slideRight();
  }

  clickHandlerLeft(e) {
    this.slideLeft();
  }

  render() {
    let { currentIndex } = this.state;
    let { listings } = this.props;
    return (
      <div className="outerCarouselComponent">
          <div className="outerLeftButton" onClick={this.clickHandlerLeft}></div>
            <div className="outerCarousel">
              <Animate translateX={this.getOffset()} tension={200} clamp>
                <div className="itemSlider">
                  {_.map(listings, (listing, index) => (
                    <OuterCarouselItem 
                      currentIndex={currentIndex}
                      listing={listing} 
                      key={JSON.stringify(index)} 
                      index={index} 
                    />
                  ))}
                </div>
              </Animate>
            </div>
          <div className="outerRightButton" onClick={this.clickHandlerRight}></div>
      </div>
    );
  };
};

export default OuterCarousel;