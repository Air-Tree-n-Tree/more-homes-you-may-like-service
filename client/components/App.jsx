import React from 'react';
import OuterCarousel from './OuterCarousel';


class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      listings: [
        {id: 0, colors: ['#bbaadd', '#bbddcc', '#aabbdd']},
        {id: 1, colors: ['#bbaacc', '#bbccaa', '#aaccdd']},
        {id: 2, colors: ['#bbccdd', '#ccddaa', '#aabbdd']},
        {id: 3, colors: ['#ccaadd', '#bbddaa', '#aabbcc']},
        {id: 4, colors: ['#bbaadd', '#bbddcc', '#aaccdd']},
        {id: 5, colors: ['#bbaacc', '#bbccaa', '#ccbbdd']},
      ]
    };
  };

  render() {
    const { listings } = this.state;
    return (
    <div>
      <h1>More homes you may like</h1>
      <OuterCarousel listings={listings}/>
    </div>
    );
  };
};

export default App;