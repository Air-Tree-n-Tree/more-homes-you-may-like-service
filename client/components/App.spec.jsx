import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import OuterCarousel from './OuterCarousel.jsx';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('This should be true', () => {

  test('adds 1 + 2 to equal 3', () => {
    expect(true).toBe(true);
  });

  
  test('It renders correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

});


describe('<App /> rendering', () => {
  let wrapper = shallow(<App />);
  it('Should find 1 <OuterCarousel /> component', () => {
    expect(wrapper.find('OuterCarousel')).toHaveLength(1);
  });
});


describe('<OuterCarousel /> rendering', () => {
  let listings = [
    {id: 0, colors: ['#bbaadd', '#bbddcc', '#aabbdd']},
    {id: 1, colors: ['#bbaacc', '#bbccaa', '#aaccdd']},
    {id: 2, colors: ['#bbccdd', '#ccddaa', '#aabbdd']},
    {id: 3, colors: ['#ccaadd', '#bbddaa', '#aabbcc']},
    {id: 4, colors: ['#bbaadd', '#bbddcc', '#aaccdd']},
    {id: 5, colors: ['#bbaacc', '#bbccaa', '#ccbbdd']},
  ];
  let wrapper = shallow(<OuterCarousel listings={listings}/>);

  it('should render 6 <OuterCarouselItem /> component', () => {
    expect(wrapper.find('OuterCarouselItem')).toHaveLength(6);
  });

});