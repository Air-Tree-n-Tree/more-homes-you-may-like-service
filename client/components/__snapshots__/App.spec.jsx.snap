// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This should be true It renders correctly 1`] = `
<div>
  <h1>
    More homes you may like
  </h1>
  <div
    className="outerCarouselComponent"
  >
    <div
      className="outerLeftButton"
      onClick={[Function]}
    />
    <div
      className="outerCarousel"
    >
      <div
        className="slider"
      >
        <span
          className="outerCarouselItem"
          style={
            Object {
              "background": "#bbaadd",
            }
          }
        >
          Outer Carousel Item Number 
          0
          <div>
            I'm a stupid innerCarousel
          </div>
        </span>
        <span
          className="outerCarouselItem"
          style={
            Object {
              "background": "#bbaacc",
            }
          }
        >
          Outer Carousel Item Number 
          1
          <div>
            I'm a stupid innerCarousel
          </div>
        </span>
        <span
          className="outerCarouselItem"
          style={
            Object {
              "background": "#bbccdd",
            }
          }
        >
          Outer Carousel Item Number 
          2
          <div>
            I'm a stupid innerCarousel
          </div>
        </span>
        <span
          className="outerCarouselItem"
          style={
            Object {
              "background": "#ccaadd",
            }
          }
        >
          Outer Carousel Item Number 
          3
          <div>
            I'm a stupid innerCarousel
          </div>
        </span>
        <span
          className="outerCarouselItem"
          style={
            Object {
              "background": "#bbaadd",
            }
          }
        >
          Outer Carousel Item Number 
          4
          <div>
            I'm a stupid innerCarousel
          </div>
        </span>
        <span
          className="outerCarouselItem"
          style={
            Object {
              "background": "#bbaacc",
            }
          }
        >
          Outer Carousel Item Number 
          5
          <div>
            I'm a stupid innerCarousel
          </div>
        </span>
      </div>
    </div>
    <div
      className="outerRightButton"
      onClick={[Function]}
    />
  </div>
</div>
`;
